source/val/validate_decorations.cpp

```diff
@@ -840,6 +840,72 @@ spv_result_t CheckDecorationsOfBuffers(ValidationState_t& vstate) {
   return SPV_SUCCESS;
 }
 
+spv_result_t CheckDecorationsOfConversions(ValidationState_t& vstate) {
+  for (const auto& kv : vstate.id_decorations()) {
+    const uint32_t id = kv.first;
+    const auto& decorations = kv.second;
+    if (decorations.empty()) {
+      continue;
+    }
+
+    const Instruction* inst = vstate.FindDef(id);
+    assert(inst);
+
+    // Validates FPRoundingMode decoration
+    for (const auto& decoration : decorations) {
+      if (decoration.dec_type() != SpvDecorationFPRoundingMode) {
+        continue;
+      }
+
+      // Validates width-only conversion instruction for floating-point object
+      // i.e., OpFConvert
+      if (inst->opcode() != SpvOpFConvert) {
+        return vstate.diag(SPV_ERROR_INVALID_ID, inst)
+               << "FPRoundingMode decoration can be applied only to a "
+                  "width-only conversion instruction for floating-point "
+                  "object.";
+      }
+
+      // Validates Object operand of an OpStore
+      bool exist = false;
+      for (const auto& store : vstate.ordered_instructions()) {
+        if (store.opcode() == SpvOpStore) {
+          const auto object_id = store.GetOperandAs<uint32_t>(1);
+          if (object_id == inst->id()) {
+            const auto ptr_id = store.GetOperandAs<uint32_t>(0);
+            const auto ptr = vstate.FindDef(ptr_id);
+            const auto ptr_type_id = ptr->GetOperandAs<uint32_t>(0);
+            const auto ptr_type = vstate.FindDef(ptr_type_id);
+            const auto storage = ptr_type->GetOperandAs<uint32_t>(1);
+
+            // Validates storage class of the pointer to the OpStore
+            if (storage != SpvStorageClassStorageBuffer &&
+                storage != SpvStorageClassUniform &&
+                storage != SpvStorageClassPushConstant &&
+                storage != SpvStorageClassInput &&
+                storage != SpvStorageClassOutput) {
+              return vstate.diag(SPV_ERROR_INVALID_ID, inst)
+                     << "FPRoundingMode decoration can be applied only to the "
+                        "Object operand of an OpStore in the StorageBuffer, "
+                        "Uniform, PushConstant, Input, or Output Storage "
+                        "Classes.";
+            }
+            exist = true;
+            break;
```

Comment:
What happens if this conversion is used by multiple stores? It seems only the first would be checked.